ui/app/models/secret-engine.js

```diff
@@ -29,8 +28,16 @@ export default SecretEngineModel.extend({
   description: attr('string', {
     editType: 'textarea',
   }),
-  config: fragment('mount-config', { defaultValue: {} }),
-  options: fragment('mount-options', { defaultValue: {} }),
+  // will only have value for kv type
+  version: attr('number', {
+    label: 'Version',
+    helpText:
+      'The KV Secrets Engine can operate in different modes. Version 1 is the original generic Secrets Engine the allows for storing of static key/value pairs. Version 2 added more features including data versioning, TTLs, and check and set.',
+    possibleValues: [2, 1],
+    // This shouldn't be defaultValue because if no version comes back from API we should assume it's v1
+    defaultFormValue: 2, // Set the form to 2 by default
+  }),
```

Comment:
Since the `mount-options` fragment only had one attribute I moved it here instead of converting to a model.